src/coreComponents/constitutive/fluid/BlackOilFluid.hpp

```diff
@@ -570,36 +560,19 @@ BlackOilFluid::KernelWrapper::
 
   // 1. Make everything zero first
 
-  for( integer ip = 0; ip < NP_BO; ++ip )
-  {
-    phaseFraction.value[ip] = 0.;
-    if( needDerivs )
-    {
-      phaseFraction.dPres[ip] = 0.0;
-    }
-    for( integer ic = 0; ic < NC_BO; ++ic )
-    {
-      phaseCompFraction.value[ip][ic] = 0.0;
-      if( needDerivs )
-      {
-        phaseFraction.dComp[ip][ic] = 0.;
-        phaseCompFraction.dPres[ip][ic] = 0.0;
-        for( integer jc = 0; jc < NC_BO; ++jc )
-        {
-          phaseCompFraction.dComp[ip][ic][jc] = 0.0;
-        }
-      }
-    }
-  }
+  LvArray::forValuesInSlice( phaseFraction.value, []( real64 & val ){ val = 0.0; } );
+  LvArray::forValuesInSlice( phaseFraction.derivs, []( real64 & val ){ val = 0.0; } );
+  LvArray::forValuesInSlice( phaseCompFraction.value, []( real64 & val ){ val = 0.0; } );
+  LvArray::forValuesInSlice( phaseCompFraction.derivs, []( real64 & val ){ val = 0.0; } );
```

Comment:
Yes, that's the best we can do without making changes to `LvArray` (which is wise to avoid for this PR), as slices do not define assignment operators, intentionally or not.